### PR TITLE
Update INSTALL.adoc

### PR DESCRIPTION
**Pull request by shindere**
This PR addresses #8658. In its current state there is
probably a lot that still needs to be done before merging.
Indeed, the documentation felt quite outdated. Also, it was not obvious
to me how far I should go, but pushes to the branch and comments
are warmly welcome. In particular, should we perhaps add an appendix to
`INSTALL.adoc` with the content of #8687?

### CODE REVIEW

File: INSTALL.adoc [hunk 1]
```diff
@@ -241,33 +114,19 @@ fairly verbose; consider redirecting the output to a file:
...
-
-7. After installation, do *not* strip the `ocamldebug` and `ocamlbrowser`
-   executables. These are mixed-mode executables (containing both compiled C
+After installation, do *not* strip the `ocamldebug` executables.
+   This is a mixed-mode executable (containing both compiled C
    code and OCaml bytecode) and stripping erases the bytecode!  Other
    executables such as `ocamlrun` can safely be stripped.
 
-== IF SOMETHING GOES WRONG
+== If something goes work
```
Comment (Gbury) [inline INSTALL.adoc]: typo: "work" -> "wrong" i suppose.

**Comment by shindere** [comment]
Guillaume Bury (2019/05/21 07:18 -0700):
> Gbury commented on this pull request.
> 
> 
> 
> >     code and OCaml bytecode) and stripping erases the bytecode!  Other
>     executables such as `ocamlrun` can safely be stripped.
>  
> -== IF SOMETHING GOES WRONG
> +== If something goes work
> 
> typo: "work" -> "wrong" i suppose.

Oh! Fixed, thanks a lot!

File: INSTALL.adoc [hunk 2]
```diff
@@ -19,166 +19,33 @@
...
-        from the compiler distribution.
-
-`-partialld <linker and options>`:: (default: determined automatically)
-        The linker and options to use for producing an object file (rather than
-        an executable) from several other object files.
-
-`-no-cfi`::
-        Do not compile support for CFI directives.
+Some options or variables like (LDFLAGS) may not be taken into account
+by the OCaml build system at the moment. Please report an issue if you
```
Comment (gasche) [inline INSTALL.adoc:36]: is the sentence cut before the end here?

**Review by gasche — COMMENTED** [review]
Looks good to me, thanks! See inline comment for a question -- it looks like a sentence is not finished.

**Comment by shindere** [comment]
Thanks! Don't know what happened with this sentence, but I fixed it.

**Review by gasche — APPROVED** [review]

**Comment by shindere** [comment]
Thanks a lot, @gasche!

I didn't expect this to be so easy! I thought there would be much more
discussion!

The life of an OCaml developer is full of surprises... :)

**Comment by shindere** [comment]
Gabriel Scherer (2019/05/21 13:51 -0700):
> Merged #8688 into trunk.

4.08: 2c4654069f47a1b5e5ebf2c2f7510aa2238e36d3
4.09: a948d3779c7fbba608193d6181aadb89599107a2

**Pull request merged**